tests/ui/stubs.js

```diff
@@ -0,0 +1,227 @@
+var fs = require("fs");
+var http = require("http");
+var url = require("url");
+
+var httputil = require("selenium-webdriver/http/util");
+var portprober = require("selenium-webdriver/net/portprober");
+var promise = require("selenium-webdriver").promise;
+
+var stubdata = require("./stubdata");
+
+function Stubs() {
```

Comment:
I would love some prose on what this is stubbing and how it works. I find that the test rig is the most important part because you need to grok it to write good tests, and to debug it. Currently I don't feel that this class is adequately explaining what its purpose is, and how it achieves said purpose. I see it does some HTTP stuff, and I guess is stubbing request/responses?